File: src/server/lib/java/org/meet4xmas/wire/IServiceAPI.java
package org.meet4xmas.wire;

public interface IServiceAPI {
	public Response registerAccount(String userId);
  public Response registerAccount(String userId, NotificationServiceInfo notificationServiceInfo); // new in version 2
  public Response deleteAccount(String userId);
  public Response createAppointment(String userId, int travelType, Location location, String[] invitees, int locationType, String userMessage);
  public Response getAppointment(int appointmentId);
  public Response getTravelPlan(int appointmentId, int travelType, Location location);
  public Response joinAppointment(int appointmentId, String userId, int travelType, Location location);
  public Response declineAppointment(int appointmentId, String userId);
  public Response finalizeAppointment(int appointmentId);
}

File: src/wp7/Meet4Xmas/org.meet4xmas.wire/IServiceAPI.cs

namespace org.meet4xmas.wire
{
    public interface IServiceAPI
    {
        Response registerAccount(string userId, object ignored /*TODO*/);
        Response deleteAccount(string userId);
        Response createAppointment(string userId, int travelType, Location location, string[] invitees, int locationType, string userMessage);
        Response getAppointment(int appointmentId);
        Response getTravelPlan(int appointmentId, int travelType, Location location);
        Response joinAppointment(int appointmentId, string userId, int travelType, Location location);
        Response declineAppointment(int appointmentId, string userId);
        Response finalizeAppointment(int appointmentId);
    }
}
